File: chanceapi-java/src/main/java/ru/dverkask/chance/Category.java
package ru.dverkask.chance;

import java.util.ArrayList;
import java.util.List;

public class Category<T> {
    private final String          name;
    private final double          weight;
    private final List<Chance<T>> items;

    public Category(String name, double weight) {
        this.name = name;
        this.weight = weight;
        this.items = new ArrayList<>();
    }

    public Category(double weight) {
        this(null, weight);
    }

    public void addItem(T item, double chance) {
        this.items.add(new Chance<>(item, chance));
    }

    public void addItem(T item) {
        this.addItem(item, weight);
    }

    public String getName() {
        return this.name;
    }

    public double getWeight() {
        return this.weight;
    }

    public List<Chance<T>> getItems() {
        return this.items;
    }
}

File: chanceapi-kotlin/src/main/kotlin/ru/dverkask/chance/Category.kt
package ru.dverkask.chance

class Category<T>(
    val name: String? = null,
    val weight: Double
) {
    private val items: MutableList<Chance<T>> = mutableListOf()

    fun addItem(item: T, chance: Double = weight) {
        items.add(Chance(item, chance))
    }

    fun getItems(): List<Chance<T>> = items
}